spring-vault-core/src/main/java/org/springframework/vault/authentication/AppRoleAuthentication.java

```diff
@@ -91,6 +96,48 @@ private VaultToken createTokenUsingAppRole() {
 		}
 	}
 
+	private String getRoleId() {
+		String roleId = options.getRoleId();
+		if (StringUtils.isEmpty(roleId) && !StringUtils.isEmpty(options.getAppRole())) {
+			try {
+				ResponseEntity<VaultResponse> response = restOperations.exchange("auth/approle/role/{role}/role-id",
+						HttpMethod.GET, createHttpEntityWithToken(), VaultResponse.class, options.getAppRole());
+				roleId = (String) response.getBody().getData().get("role_id");
+			} catch (HttpStatusCodeException e) {
+				throw new VaultException(String.format("Cannot get Role id using AppRole: %s",
+					VaultResponses.getError(e.getResponseBodyAsString())));
+			}
+		}
+		return roleId;
+
+	}
+
+	private String getSecretId() {
+		String secretId = options.getSecretId();
+		if(StringUtils.isEmpty(secretId) && !StringUtils.isEmpty(options.getAppRole())) {
```

Comment:
We should assume that `appRole` is set otherwise this case could return an empty SecretId.